src/pytest_benchmark/plugin.py

```diff
@@ -857,7 +866,10 @@ def display_results_table(self, tr):
                 worst[prop] = max(benchmark[prop] for _, benchmark in report_progress(
                     benchmarks, tr, "{line} ({pos}/{total})", line=line))
 
-            unit, adjustment = time_unit(best.get(self.sort, benchmarks[0][self.sort]))
+            time_unit_key = self.sort
+            if self.sort in ("name", "fullname"):
```

Comment:
This made me realize I lumped up two distinct things in `--benchmark-sort`:

* the actual sort
* what column to pick the unit from

What if you have `--benchmark-sort=name` and you want the unit from `mean`? Is this realistic, what do you think? It would mean yet another option :confused:  ...